investigations/LOKI/loki_performance.md

```diff
@@ -0,0 +1,178 @@
+
+## Hardware 
+
+The hardware used for these tests is as follows:
+
+- 128GB RAM
+- Intel(R) Core(TM) i9-9920X CPU @ 3.50GHz
+- Target Linux Platform (OpenMP is a hard requirement)
+
+## Outcomes
+
+### Measurement Interpretation
+The ESS pulse rate for neutrons is expected to be **14Hz**. This will be the rate at which messaged will be produced in the Kafka system and therefore consumed by Mantid. Measurements are taken in this context. If Mantid is capable of processing events (writing events to in-memory data structures and performing basic reduction) at this rate at a target neutron flux, this indicates that Mantid can cope with the event rate.
+
+### Data ingest into Mantid
+
+Data consumption into Mantid using realistic SANS data generated using GEANT4 resulted in an average consumption rate of ~24Hz. This was performed at an event flux of ~1.4e<sup>7</sup> events per message. This clearly shows that Mantid can comfortably deal with the ingest of data at these high rates.
+
+##### Scaling
+There was previous work done in investigating how well the live streamer in Mantid scales and was presented in this document [LiveReductionInvestigation](https://github.com/DMSC-Instrument-Data/documents/blob/master/investigations/Live%20Reduction/LiveReductionInvestigation.md). Scaling for LOKI ingesting the simulated data showed the following trend:
+
+`#` of threads|Message Consumption Rate (Hz)
+---|---
+1|8
+4|26
+8|33
+16|52
+24|21
+
+Note that at 24 threads it is assumed the host system is oversubscribed due to running Mantid, Kafka broker and NeXus publisher simultaneously as well as hyperthreading.
+
+### SANS Data Reduction
+The Data reduction used for these performance investigations was the ISIS SANS Reduction workflow which was decided in 2018 as part of the DMSC milestones (see [here](https://confluence.esss.lu.se/pages/viewpage.action?pageId=262411283)). Profiling on this workflow was performed by Neil Vaytet (ESS)> The data for this test was produced using Geant4 and converted into a Mantid data structure for processing. The results are shown below:
+
+![Sans Reduction Profile](sansreductionprofile.png)
+
+In this test, 12 CPU cores were allocated to the data reduction, we see the reduction is mostly using all allocated cores most of the time. This suggests this workflow is currently optimised to make use of parallel CPU architectures. However, `ConvertUnits` seems to be entirely single threaded and half of the `Rebin` operation is single-threaded. Recommendations for optimizing the workflow are presented in the last section of this document.
+
+### Ingest with ISIS SANS Workflow
+
+The ingest of data into Mantid was coupled with the ISIS SANS Workflow to assess the impact on the message consumption rate. Mantid reported an average consumption rate of ~18Hz. This is still above the 14Hz target but there is the possibility of optimizing the SANS workflow algorithms to obtain better performance.
```

Comment:
Great. This is one of the most important outcomes of the report as far as the readers will be concerned.